prov/rxm/src/rxm_domain.c

```diff
@@ -589,6 +597,20 @@ static int rxm_mr_regattr(struct fid *fid, const struct fi_mr_attr *attr,
 	rxm_mr->device = msg_attr.device.reserved;
 	*mr = &rxm_mr->mr_fid;
 
+	// get the GDR copy registration if possible, inspired from
+	// efa_mr_hmem_setup
+	if (cuda_is_gdrcopy_enabled()) {
+		int gdrerr = ofi_hmem_dev_register(
+			rxm_mr->iface, attr->mr_iov->iov_base,
```

Comment:
I think for @j-xiong's comment ^^, shouldn't only do this whole thing in a if block of `if (rxm_mr->iface == FI_HMEM_CUDA)`? Then it's less concerning to me that you replace `rxm_mr->iface` to `FI_HMEM_CUDA` within such block

Reply:
We have a good chance to enable other ifaces here. Limiting it to cuda only is less desirable.